### DISCUSSION TITLE
Self-hosting `umami.js` among website’s files (hosting in the same domain)

### DISCUSSION THREAD

**Discussion by xplosionmind** [umami-software/umami#1797] [Q&A]
Hello, the domain where I host my umami has been added to uBlock Origin blocklist. I would like to avoid this by hosting the `umami.js` file among my website’s files, therefore having it in the same and identical domain as my website.

Would the script work anyway?

**Comment by orestbida** [comment]
Yes, it should work just fine. Just remember to also update the tracking script if you upgrade umami.

**Reply by xplosionmind** [reply]
Thanks a lot! I do not have to change anything in the script?

**Reply by orestbida** [reply]
The [script](https://github.com/umami-software/umami/blob/master/tracker/index.js) is just vanilla js, and does not need to be modified/rebuilt.

However, it won't work properly on local development as it attempts to send a `post` request to the domain where it was initially loaded from, such as `localhost/api/...`.

To fix this, you need to manually specify the correct domain by using the [`data-host-url`](https://umami.is/docs/tracker-configuration) attribute.
